src/melee.cpp

```diff
@@ -1590,27 +1634,32 @@ bool Character::block_hit( Creature *source, body_part &bp_hit, damage_instance
 
     float total_damage = 0.0;
     float damage_blocked = 0.0;
+
     for( auto &elem : dam.damage_units ) {
         total_damage += elem.amount;
+
         // block physical damage "normally"
         if( elem.type == DT_BASH || elem.type == DT_CUT || elem.type == DT_STAB ) {
             // use up our flat block bonus first
             float block_amount = std::min( total_phys_block, elem.amount );
             total_phys_block -= block_amount;
             elem.amount -= block_amount;
             damage_blocked += block_amount;
+
             if( elem.amount <= std::numeric_limits<float>::epsilon() ) {
                 continue;
             }
 
             float previous_amount = elem.amount;
             elem.amount *= physical_block_multiplier;
             damage_blocked += previous_amount - elem.amount;
-            // non-electrical "elemental" damage types do their full damage if unarmed,
-            // but severely mitigated damage if not
-        } else if( elem.type == DT_HEAT || elem.type == DT_ACID || elem.type == DT_COLD ) {
+        }
+
+        // non-electrical "elemental" damage types do their full damage if unarmed,
+        // but severely mitigated damage if not
+        else if( elem.type == DT_HEAT || elem.type == DT_ACID || elem.type == DT_COLD ) {
             // Unarmed weapons won't block those
-            if( !martial_arts_data.is_force_unarmed() && weapon_blocking && !unarmed ) {
+            if( !force_unarmed && item_blocking && !unarmed ) {
```

Comment:
since you use `!force_unarmed && item_blocking && !unarmed` in a couple of places and it is not immediately obvious what it means, would you please break it out into a `const bool` local variable with a meaningful name?

Reply:
I renamed item_blocking to has_shield since it's more appropriate, and named a new const bool local var to item_blocking since ultimately ```!force_unarmed && item_blocking && !unarmed``` decides whether or not a block is being performed while unarmed.